### PR TITLE
CA1032 in IotHubException class

### PR DESCRIPTION
**Pull request by abhipsaMisra**
CA1032: Implement standard exception constructors
Cause: A type extends System.Exception but doesn't declare all the required constructors.

https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1032

Also added XML comments for public methods and properties (CS1591).

### CODE REVIEW

File: iothub/service/src/Common/Exceptions/IotHubException.cs [hunk 1]
```diff
@@ -6,88 +6,180 @@
...
+                TrackingId = info.GetString(TrackingIdSerializationStoreName);
             }
         }
 
+        /// <summary>
+        /// Sets the <see cref="IsTransient"/> and <see cref="TrackingId"/> information to the <see cref="SerializationInfo"/>.
+        /// </summary>
+        /// <param name="info">The <see cref="SerializationInfo"/> that holds the serialized object data about the exception being thrown.</param>
+        /// <param name="context">The <see cref="StreamingContext"/> that contains contextual information about the source or destination.</param>
         public override void GetObjectData(SerializationInfo info, StreamingContext context)
```
Comment (abhipsaMisra) [inline iothub/service/src/Common/Exceptions/IotHubException.cs:173]: From https://docs.microsoft.com/en-us/dotnet/api/system.exception.getobjectdata?view=net-5.0

File: iothub/service/src/Common/Exceptions/IotHubException.cs [hunk 2]
```diff
@@ -6,88 +6,180 @@
...
+            TrackingId = trackingId;
         }
 
+        /// <summary>
+        /// Creates an instance of <see cref="IotHubException"/> with the <see cref="SerializationInfo"/>
+        /// and <see cref="StreamingContext"/> associated with the exception.
+        /// </summary>
+        /// <param name="info">The <see cref="SerializationInfo"/> that holds the serialized object data about the exception being thrown.</param>
+        /// <param name="context">The <see cref="StreamingContext"/> that contains contextual information about the source or destination.</param>
         protected IotHubException(SerializationInfo info, StreamingContext context)
```
Comment (abhipsaMisra) [inline iothub/service/src/Common/Exceptions/IotHubException.cs:158]: From https://docs.microsoft.com/en-us/dotnet/api/system.exception.-ctor?view=net-5.0#System_Exception__ctor_System_Runtime_Serialization_SerializationInfo_System_Runtime_Serialization_StreamingContext_

File: iothub/service/src/Common/Exceptions/IotHubException.cs [hunk 3]
```diff
@@ -6,88 +6,180 @@
...
+    /// <summary>
+    /// The exception thrown when the client receives an error while communicating with IoT Hub service.
+    /// </summary>
     [Serializable]
     public class IotHubException : Exception
     {
+        private const string IotHubExceptionDefaultMessage = "An IoT Hub exception has occurred.";
         private const string IsTransientValueSerializationStoreName = "IotHubException-IsTransient";
         private const string TrackingIdSerializationStoreName = "IoTHubException-TrackingId";
-        private const string ErrorCodeName = "ErrorCode";
```
Comment (abhipsaMisra) [inline iothub/service/src/Common/Exceptions/IotHubException.cs:14]: this was unused

File: iothub/service/src/Common/Exceptions/IotHubException.cs [hunk 4]
```diff
@@ -6,88 +6,180 @@
...
+        /// </summary>
         public bool IsTransient { get; private set; }
 
+        /// <summary>
+        /// The service returned tracking Id associated with this particular error.
+        /// </summary>
         public string TrackingId { get; set; }
 
+        /// <summary>
+        /// Create an instance of <see cref="IotHubException"/> with the default error message and mark it as non-transient.
```
Comment (drwill-ms) [inline iothub/service/src/Common/Exceptions/IotHubException.cs:30]: Creates (although there is no subject in this sentence, the implied subject is 'It' which would make the conjugation of **to create** as **creates**).

**Review by drwill-ms — APPROVED** [review]

File: iothub/service/src/Common/Exceptions/IotHubException.cs (hunk 4 above)
Comment (abhipsaMisra) [reply iothub/service/src/Common/Exceptions/IotHubException.cs:30]: will do!

**Review by azabbasi — APPROVED** [review]

**Pull request merged**